src/components/editgrid/EditGrid.js

```diff
@@ -941,6 +941,13 @@ export default class EditGridComponent extends NestedArrayComponent {
       this.setNestedValue(component, editRow.data, flags);
     });
   }
+
+  setPristine(pristine) {
+    super.setPristine(pristine);
+    if (pristine) {
+      this.editRows = [];
```

Comment:
This may cause unintended consequences. You are making a bad assumption that the only time that pristine is set is when the EditGrid is canceled. I can think of many cases when pristine is set to true when this is not the case.